CMake/AbseilHelpers.cmake

```diff
@@ -62,7 +62,114 @@ function(absl_library)
   endif()
 endfunction()
 
+#
+# CMake function to imitate Bazel's cc_library rule.
+#
+# Parameters:
+# NAME: name of target (see Note)
+# HDRS: List of public header files for the library
+# SRCS: List of source files for the library
+# DEPS: List of other libraries to be linked in to the binary targets
+# COPTS: List of private compile options
+# DEFINES: List of public defines
+# LINKOPTS: List of link options
+# VISIBILITY_PUBLIC: Add this so that this library will be exported under absl:: (see Note).
+# TESTONLY: When added, this target will only be built if user passes -DBUILD_TESTING=ON to CMake.
+#
+# Note:
+#
+# By default, absl_cc_library will always create a library named absl_${NAME},
```

Comment:
`absl_internal_` preffix will reduce the chance of name collision, but we might still face name collision *within* Abseil itself (some targets have name that are too general like `test_util` in `//absl/time` package). I have a longer response at https://groups.google.com/forum/#!topic/abseil-io/_Pq2NuCuYcc

I am settling at `absl_` preffix for now since Abseil's CMake files use `absl_` preffix too (e.g. `absl_throw_delegate`), so I don't have to do too much manual editing when showing some possible migrations in this PR.

Reply:
I'm not really worried about name collisions as much as I'm trying to make it as inconvenient as possible to rely on abseil internals, since we have no true hiding mechanism as we do in Bazel.  Does that make more sense?

Reply:
Done.